api/src/main/java/edu/cornell/mannlib/vitro/webapp/audit/controller/AuditController.java

```diff
@@ -0,0 +1,334 @@
+import edu.cornell.mannlib.vitro.webapp.auth.permissions.PermissionSets;
+import edu.cornell.mannlib.vitro.webapp.auth.permissions.SimplePermission;
+import edu.cornell.mannlib.vitro.webapp.auth.requestedAction.AuthorizationRequest;
+import edu.cornell.mannlib.vitro.webapp.beans.UserAccount;
+import edu.cornell.mannlib.vitro.webapp.controller.VitroRequest;
+import edu.cornell.mannlib.vitro.webapp.controller.freemarker.FreemarkerHttpServlet;
+import edu.cornell.mannlib.vitro.webapp.controller.freemarker.UrlBuilder;
+import edu.cornell.mannlib.vitro.webapp.controller.freemarker.responsevalues.ResponseValues;
+import edu.cornell.mannlib.vitro.webapp.controller.freemarker.responsevalues.TemplateResponseValues;
+import edu.cornell.mannlib.vitro.webapp.dao.UserAccountsDao;
+import edu.cornell.mannlib.vitro.webapp.modelaccess.ModelAccess;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.time.DateUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * UI for browsing audit entries
+ */
+@WebServlet(name = "AuditViewer", urlPatterns = { "/audit/*" })
+public class AuditController extends FreemarkerHttpServlet {
+    private static final Log log = LogFactory.getLog(AuditController.class);
+
+    // Template for the UI
+    private static final String TEMPLATE_DEFAULT = "auditHistory.ftl";
+
+    private static final String PARAM_OFFSET = "offset";
+    private static final String PARAM_LIMIT = "limit";
+    private static final String PARAM_GRAPH = "graph";
+    private static final String PARAM_ORDER = "order";
+    private static final String PARAM_START_DATE = "start_date";
+    private static final String PARAM_END_DATE = "end_date";
+    private static final String PARAM_USER_URI = "user";
+    private static final String DESC_ORDER = "DESC";
+    private static final String ASC_ORDER = "ASC";
+    private static final String[] limits = { "10", "30", "50", "100", "1000" };
+    private static final String[] orders = { ASC_ORDER, DESC_ORDER };
+
+    private final SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+
+    @Override
+    protected AuthorizationRequest requiredActions(VitroRequest vreq) {
+        return SimplePermission.USE_MISCELLANEOUS_ADMIN_PAGES.ACTION;
+    }
+
+    @Override
+    protected ResponseValues processRequest(VitroRequest vreq) {
+        if (log.isDebugEnabled()) {
+            dumpRequestParameters(vreq);
+        }
+        UserAccountsDao uad = ModelAccess.on(vreq).getWebappDaoFactory().getUserAccountsDao();
+        Map<String, Object> body = new HashMap<>();
+
+        // Get the current user
+        UserAccount acc = LoginStatusBean.getCurrentUser(vreq);
+        if (acc == null || !isAdmin(acc)) {
+            return new TemplateResponseValues("error-standard.ftl", body);
+        }
+        // Get the offset parameter (or default if unset)
+        int offset = getOffset(vreq);
+        body.put("offset", offset);
```

Comment:
Why use the string literal when the constant is above?

Reply:
fixed